tests/Cache/CachedApiKeyAuthenticatorTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Cache;
+
+use CultuurNet\UDB3\ApiGuard\ApiKey\ApiKey;
+use CultuurNet\UDB3\ApiGuard\ApiKey\ApiKeyAuthenticationException;
+use CultuurNet\UDB3\ApiGuard\ApiKey\ApiKeyAuthenticator;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Cache\Adapter\ArrayAdapter;
+
+final class CachedApiKeyAuthenticatorTest extends TestCase
+{
+    /**
+     * @var ApiKeyAuthenticator&MockObject
+     */
+    private $fallbackApiKeyAuthenticator;
+
+    private ApiKey $cachedApiKey;
+
+    private CachedApiKeyAuthenticator $cachedApiKeyAuthenticator;
+
+    protected function setUp(): void
+    {
+        $this->fallbackApiKeyAuthenticator = $this->createMock(ApiKeyAuthenticator::class);
+        $cache = new ArrayAdapter();
+        $this->cachedApiKey = new ApiKey('b26e5a7b-5e01-46c1-8da8-f45edc51d01a');
+
+        $this->cachedApiKeyAuthenticator = new CachedApiKeyAuthenticator(
+            $this->fallbackApiKeyAuthenticator,
+            $cache
+        );
+
+        $cache->get(
+            $this->cachedApiKey->toString(),
+            function () {
+                return false;
+            }
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_can_get_a_cached_api_key(): void
+    {
+        $this->fallbackApiKeyAuthenticator->expects($this->never())
+            ->method('authenticate');
+
+        $this->expectException(ApiKeyAuthenticationException::class);
```

Comment:
Nice to see all these tests!

But I'm a bit confused by this exception. Does that mean only a key that is not authorised is tested? If so, would it make sense to also test a key that is authorised and doesn't throw an exception?